src/io/iohk/decco/test/snappy/SnappySpec.scala

```diff
@@ -0,0 +1,25 @@
+package io.iohk.decco.snappy
+
+import io.iohk.decco.test.utils.CodecTestingHelpers._
+import io.iohk.decco.definitions._
+import io.iohk.decco.snappy.auto._
+import org.scalatest.{FlatSpec, Matchers}
+import io.iohk.decco.BufferInstantiator.global.HeapByteBuffer
+
+class SnappySpec extends FlatSpec with Matchers {
+  behavior of "snappy"
+
+  it should "encode data properly" in {
+    testCodec[Array[Byte]]
+    testCodec[String]
+  }
+
+  it should "compress message" in {
+    val compressableString = (1 to 100).map(_ => "foo bar").mkString("start", " ", "end")
+
+    val naiveCodec = io.iohk.decco.auto.codecContract2Codec[String]
```

Comment:
Did you mean 'native' or 'naive'?

Reply:
I've meant naive understood as "having or showing unaffected simplicity of nature; unsophisticated; ingenuous". If the wording feels weird here, I can change it (would `simpleCodec` or `directCodec be ok?).